### ISSUE TITLE
Feature request: Allow to search archived notes as well

### ISSUE THREAD

**Issue by retgits** [drgrib/alfred-bear#65]
Let me start by thanking you for this workflow, it's really awesome! 

In Bear, I currently use a tag called "archive" as opposed to using the archive functionality in Bear itself. As a feature request, I'd like the ability to have the workflow search archived notes as well. Personally, I can see two options:

1. Remove `note.ZARCHIVED = 0` from the queries to always search archived notes
2. Add a variable that determines whether to search archived notes or not

I'd be happy to help out and contribute :)

**Comment by drgrib** [comment]
Option 2 seems reasonable to me, though it is more work. It reminds me that I want to put query size (the `LIMIT` value in the SQL queries) into a variable as well.

I'm tempted to consider Option 1 but I can imagine that people would want archived notes not to show up.

That said, this also strikes me as a pretty unusual use case. May I ask why you want to archive notes but also search them? Wouldn't archiving them imply you want them separated from your usual workflow (which is the rational for the current `note.ZARCHIVED = 0` condition)?

**Comment by retgits** [comment]
> I'm tempted to consider Option 1 but I can imagine that people would want archived notes not to show up.

Agreed, I can absolutely understand that not everyone wants to see their archived notes all the time

> Wouldn't archiving them imply you want them separated from your usual workflow (which is the rational for the current note.ZARCHIVED = 0 condition)?

Yes, which leads me to a third option. Have a separate query which allows to search for archived notes.

> May I ask why you want to archive notes but also search them? 

Sure! I use Bear as my "second brain", structured using PARA (Projects, Areas, Resources, Archives). When my projects are done, I move them to an "archives" tag. While that arguably solves the use case as well, I want to get them more out of sight so I'm not distracted by them. In certain scenarios, like starting new projects or needing something specific, I revisit my archive.

**Comment by drgrib** [comment]
Interesting. I think a special keyword rather than a wholly new type of query may be more useful as a third option. That way we wouldn't need entirely new versions of `bn`, `bs`, and `bcs`.

That said, assuming [this page](https://fortelabs.com/blog/para/) is an accurate explanation of PARA, I think I already implicitly use this system myself.

The difference is that I structure all of my Projects and Archive to exist inside the [Things](https://culturedcode.com/things/) app, which I find much more natural for organizing tasks and projects that need archiving. Conveniently, projects and tasks are automatically archived as soon as they are marked completed.

I tend to see Resources and Areas as evergreen. My Resources are my notes and my Areas are my tags. I see knowledge as monotonically increasing and accumulating more can only be a good thing. These are the types of notes I store in Bear, where they live forever and are searchable using this workflow. Any note that becomes obsolete can be updated or combined with other notes, which tools like tag searching and the `bcs` keyword are meant to facilitate. My tags also sometimes need to be restructured, maybe once a year or two, which I just do within the main Bear app. I didn't even know the archive feature existed because it is a feature I never sought out and would never consider using.
